### ISSUE TITLE
Ability to attach latest version of AWS Lambda Layer

### ISSUE THREAD

**Issue by exoego** [serverless/serverless#7740]
### Use case description

https://www.serverless.com/framework/docs/providers/aws/guide/layers/

Currently, we can attach a specific version of Lambda Layer to functions like below:

```yaml
functions:
  hello:
    layers:
      - arn:aws:lambda:region:XXXXXX:layer:LayerName:Y
```

I think there are some use case that users want to use always the latest version of layer.
One example is sharing common dependencies (node_modules) among services.
Services should always use the latest version of layer.

However, we need to watch that there is new version and update version`Y` manually whenever new version `Z` is created.

### Proposed solution

```yaml
functions:
  hello:
    layers:
      - arn:aws:lambda:region:XXXXXX:layer:LayerName:LATEST
```

If serverless detects `LATEST` version specified, serverless fetch the latest version and rewrite the CloudFormation before deploying.
So whenever service is deployed, layers are automatically synced to latest version.

**Comment by medikoo** [comment]
Thanks @exoego looks definitely worthwhile.

> If serverless detects LATEST version specified, serverless fetch the latest version and rewrite the CloudFormation before deploying.

How will Framework know (just basing on lambda ARN), where to query for latest version?

**Comment by exoego** [comment]
Good question.
I am considering to use `listLayerVersions" with layer name.

**Comment by medikoo** [comment]
> I am considering to use `listLayerVersions" with layer name.

@exoego can you elaborate? `listLayerVersions` will be exposed by which entity? How we should know just by ARN where to look for it?

**Comment by exoego** [comment]
AWS Lambda Service class offers `listLayerVersions` API.
ARN contains LayerName, which can be input for the API.

**Comment by exoego** [comment]
Hmm, I prototyped with the above strategy (using LayerName with `Lambda.listLayerVersions`) and figured out that the strategy does not work if the layer is created by other AWS account (e.g. [public layers](https://github.com/mthenw/awesome-layers)). 
It means that this feature can be applied to only layers within same AWS account.
It is useful, but use-case is limited...

I expect AWS adds new API `Lambda.getLatestLayerVersion(arnOrLayerName)`

**Comment by danielsan** [comment]
At [Everymundo](https://everymundo.com) we use this [template repo](https://github.com/EveryMundo/aws-serverless-node-lambda-layer-template#how-to-use-the-published-layers-within-your-serverless-project) to create all our nodejs layers and you can see there [how to use the published layers within your serverless project](https://github.com/EveryMundo/aws-serverless-node-lambda-layer-template#how-to-use-the-published-layers-within-your-serverless-project) and always get the latest deployed version of the lib.

The caveat is that you have to create your layers using the serverless framework.

**Comment by adriantomas** [comment]
@exoego This is what I found so far for obtaining latest layer from layer name only:

```aws lambda list-layer-versions --layer-name <LayerName> --region <Region> --query 'LayerVersions[0].LayerVersionArn'```

**Comment by exoego** [comment]
@adriantomas 
Thanks for suggestion.
But I have already prototyped with the same strategy, but found that the strategy can not fetch version of layer created by other AWS accounts (see https://github.com/serverless/serverless/issues/7740#issuecomment-630179903).

@medikoo 
If the limitation (`LATEST` works for layers created within same AWS account) is fine, I will open a PR.

**Comment by adriantomas** [comment]
There's another workaround as @danielsan stated in https://github.com/serverless/serverless/issues/7740#issuecomment-634988602
```
cf:stackName.outputKey
```
![image (2)](https://user-images.githubusercontent.com/22175056/84121960-0f4fcd00-aa38-11ea-92cb-e0f413e8e4da.jpg)

**Comment by medikoo** [comment]
> If the limitation (LATEST work only for layers created within same AWS account) is fine, I will open a PR.

I think it's fine assuming that:
- Significant percentage of users that rely on layers, use layers from same AWS account (I don't know numbers on that)
- It'll remain friendly (not confusing) for those which work with layers coming from other AWS accounts

**Comment by feeh27** [comment]
You can use plugins: serverless-latest-layer-version (this is compatible with serverless-pseudo-parameters plugin)

**Comment by exoego** [comment]
Thanks !!

Do you mean this plugin ? https://www.npmjs.com/package/serverless-latest-layer-version
The plugin looks enough for current proposal.

I am closing this feature request since community plugin already covers use case.

**Comment by feeh27** [comment]
This is the plugin.

You can also use the serverless-pseudo-parameters plugin for variables like aws accountId, aws region, etc.

The two mentioned plugins are compatible.

Here is a link to the serverless-pseudo-parameters plugin:
https://www.npmjs.com/package/serverless-pseudo-parameters

**Comment by corydorning53** [comment]
So i'm running into this issue and trying to use `serverless-latest-layer-version` but it does not support the use of profiles:
https://github.com/mooyoul/serverless-latest-layer-version/issues/7

Any chance of landing this into serverless core instead?

**Comment by medikoo** [comment]
@corydorning53 I've commented at mooyoul/serverless-latest-layer-version#7 on how this issue can be fixed in a plugin

**Comment by corydorning53** [comment]
> @corydorning53 I've commented at [mooyoul/serverless-latest-layer-version#7](https://github.com/mooyoul/serverless-latest-layer-version/issues/7) on how this issue can be fixed in a plugin

yeah i noticed. plugin doesn't seem to be actively developed any longer with this being the only report issue and the last activity on it being 5 months ago. :/

**Comment by medikoo** [comment]
@corydorning53 I think we can incorporate that to the Framework. Still to not confuse things, I believe best would be if we'd rely on current variables syntax, we may try to introduce another `context` scope for that:

```yaml
functions:
  hello:
    layers:
      - arn:aws:lambda:region:XXXXXX:layer:LayerName:${context:latest}
```

**Comment by corydorning53** [comment]
In regards to the how, i'll leave it the experts 👍 

Does an open issue on this exist that i can follow for implementation?

**Comment by medikoo** [comment]
Let me reopen this one

**Comment by circulon** [comment]
Has there been any progress on this yet?
Thanks

**Comment by fedsp** [comment]
Has there any progress on this yet???

**Comment by pgrzesik** [comment]
There was no progress on this issue, however, it looks like this use case is currently addressed by the previously mentioned plugin - did you try it @fedsp ?

**Comment by medikoo** [comment]
I've updated the spec, to actually solve it with new variables resolver, and it should be pretty straightforward to do

**Comment by fedsp** [comment]
Hi, the plugin is for nodejs.

My usecase uses python with boto3.

**Comment by pgrzesik** [comment]
Hello @fedsp - it should be runtime agnostic, could you clarify what do you mean by `I was expecting something from the api/server side, not from the client.`?

**Comment by efelipe402** [comment]
Waiting for any update for this layer version

**Comment by efelipe402** [comment]
@exoego in which of serverless framwork version did you tested it out? because for me didn't work with single, multiple quotes, hardcored arn, using custom variables etc

this is the error shown

Cannot resolve serverless.yml: Variables resolution errored with:
    - Cannot resolve variable at "functions.getOrderTransactionId.layers.0": Unsupported "latest" address argument in variable "aws" source,


my env.

Your Environment Information ---------------------------
     Operating System:          darwin
     Node Version:              14.14.0
     Framework Version:         2.51.0
     Plugin Version:            5.5.1
     SDK Version:               4.3.0
     Components Version:        3.18.1

**Comment by exoego** [comment]
> in which of serverless framwork version did you tested it out?

I do not remember precisely but it was 1.x

**Comment by KobeBlockify** [comment]
Anything ?
